_studio/shared/src/libmfx_core.cpp

```diff
@@ -82,22 +82,22 @@ mfxStatus CommonCORE::API_1_19_Adapter::QueryPlatform(mfxPlatform* platform)
 
 mfxStatus CommonCORE::AllocBuffer(mfxU32 nbytes, mfxU16 type, mfxHDL *mid)
 {
-    UMC::AutomaticUMCMutex guard(m_guard);
+    std::lock_guard<std::recursive_mutex> guard(m_guard);
```

Comment:
do we really need ```recursive_mutex``` here? Using of this kind of mutexes often indicates bad design